### ISSUE TITLE
How to use SVG/images as axis ticks?

### ISSUE THREAD

**Issue by fromaline** [f5/unovis#513]
Hey, thanks for this fantastic library!

I’m using it in a Svelte/SvelteKit project and’d like to use SVG as an axis tick label. I tried using `tickFormat` to return an `<img>` tag or an SVG directly, but it doesn't work. Is there a way to do it?

**Comment by rokotyan** [comment]
Thanks @fromaline 

Axis tick labels don't support custom content. This would be a great addition to the library. Let me know if you want to try adding this feature yourself, I'll be happy to point you to the right place in the code.

**Comment by fromaline** [comment]
@rokotyan, thanks for your response!

I really need this functionality for my project, so I’ll give it a shot. Could you please point me in the right direction?

**Comment by rokotyan** [comment]
@fromaline Once you've cloned the repo, do `npm install` and then `npm run dev`. This will run our demo app.

Search for an Axis example there. Ideally copy an example to have a dedicated one for the feature you'll be working on (maybe copy this one `packages/dev/src/examples/xy-components/axis/single-axis`).

Then explore this section of the Area component's code:
https://github.com/f5/unovis/blob/75c6dfef7f74aaa3e48880e1892437da00e8b25c/packages/ts/src/components/axis/index.ts#L222-L245

Most likely your changes will go somewhere there.

**Comment by fromaline** [comment]
@rokotyan, thanks for the info!

I went through the code, and it seems highly tailored toward text-based ticks. Do you have any suggestions for implementing images as ticks? I’d like to keep it as simple as possible, but it feels like it might require quite a bit of coding and testing — I’m not sure it’s worth the effort.

**Comment by rokotyan** [comment]
@fromaline Yes, you'll need to completely replace the `<text>` element of each tick.

If I were to implement this, I would allow `tickFormat` to return a custom SVG string, then check in the code to see if the current tick value has any SVG markup. And if it does, completely replace the contents of `g.tick` with that custom markup. 

The Graph component has something similar:
https://github.com/f5/unovis/blob/75c6dfef7f74aaa3e48880e1892437da00e8b25c/packages/ts/src/components/graph/modules/shape.ts#L27-L39

**Comment by fromaline** [comment]
@rokotyan, got it! Thanks for the hint!
